src/lib/logical_query_plan/mock_node.hpp

```diff
@@ -6,28 +6,41 @@
 #include <vector>
 
 #include "abstract_lqp_node.hpp"
+#include "all_type_variant.hpp"
 
 namespace opossum {
 
 /**
- * Node that represents a table that has no data backing it, but may provide (mocked) statistics. Intended for usage in
- * tests (e.g. general LQP tests, optimizer tests that just rely on statistics and not actual data) and the playground
+ * Node that represents a table that has no data backing it, but may provide
+ *  - (mocked) statistics
+ *  - or just a column layout. It will pretend it created the columns.
+ * It is useful in tests (e.g. general LQP tests, optimizer tests that just rely on statistics and not actual data) and
+ * the playground
  */
 class MockNode : public AbstractLQPNode {
  public:
-  MockNode();
-  explicit MockNode(const std::shared_ptr<TableStatistics>& statistics);
+  using ColumnDefinitions = std::vector<std::pair<DataType, std::string>>;
+
+  explicit MockNode(const std::optional<std::string>& alias = std::nullopt);
+  explicit MockNode(const ColumnDefinitions& column_definitions,
+                    const std::optional<std::string>& alias = std::nullopt);
+  explicit MockNode(const std::shared_ptr<TableStatistics>& statistics,
+                    const std::optional<std::string>& alias = std::nullopt);
 
-  const std::vector<ColumnID>& output_column_ids_to_input_column_ids() const override;
   const std::vector<std::string>& output_column_names() const override;
 
   std::string description() const override;
   std::string get_verbose_column_name(ColumnID column_id) const override;
 
  protected:
-  std::shared_ptr<AbstractLQPNode> _deep_copy_impl() const override;
+  std::shared_ptr<AbstractLQPNode> _deep_copy_impl(const std::shared_ptr<AbstractLQPNode>& left_child,
+                                                   const std::shared_ptr<AbstractLQPNode>& right_child) const override;
 
  private:
   std::vector<std::string> _output_column_names;
+
+  // Constructor args to keep around for deep_copy()
+  std::optional<ColumnDefinitions> _column_definitions;
+  std::optional<std::shared_ptr<TableStatistics>> _table_statistics;
```

Comment:
union?

Reply:
hate dealing with variants, but as you wish.

Reply:
I don't hate them in general, they are just so clumsy...